### PR TITLE
Enable dynamic MPI in github CI when built as neuron submodule

### PR DESCRIPTION
**Pull request by pramodk**
* Add CI for dynamic MPI under NEURON
   - Currently there is no CI that builds neuron+coreneuron with dynamic MPI support
   - This should help to detect errors like below early
       https://dev.azure.com/neuronsimulator/nrn/_build/results?buildId=4649&view=logs
* Add missing include path for dynamic MPI targets
* Cherry-pick improvement from #744 

Fixes # (issue)

Azure CI failure in https://github.com/neuronsimulator/nrn/pull/1580

**How to test this?**

```bash
cmake .. -DCMAKE_INSTALL_PREFIX=`pwd`/install -DNRN_ENABLE_INTERVIEWS=OFF -DNRN_ENABLE_RX3D=OFF -DPYTHON_EXECUTABLE=`which python3` -DCORENRN_ENABLE_GPU=OFF   -DCORENRN_ENABLE_NMODL=OFF  -DCMAKE_INSTALL_PREFIX=../install -DNRN_ENABLE_CORENEURON=ON -DNRN_ENABLE_TESTS=ON -DNRN_ENABLE_MPI_DYNAMIC=ON
make -j12
```

**Use certain branches for the SimulationStack CI**

CI_BRANCHES:NEURON_BRANCH=pramodk/coreneuron-update,

### TODO

- [ ] Fix GitHub CI tests due to coreneuron dynamic MPI

Co-authored-by: Olli Lupton <oliver.lupton@epfl.ch>

### CODE REVIEW

File: .github/workflows/test-as-submodule.yml [hunk 1]
```diff
@@ -43,18 +44,30 @@ jobs:
...
             ninja-build openmpi-bin python3-dev python3-numpy python3-pytest \
-            python3-pytest-cov
+            python3-pytest-cov jq
           echo "CC=gcc" >> $GITHUB_ENV
           echo "CXX=g++" >> $GITHUB_ENV
 
+      - name: Set NEURON branch
+        id: vars
+        env:
+          GITHUB_CONTEXT: ${{ toJSON(github) }}
```
Comment (olupton) [inline .github/workflows/test-as-submodule.yml:54]:
```suggestion
          GITHUB_PR_BODY: ${{github.event.pull_request.body}}
```

File: .github/workflows/test-as-submodule.yml [hunk 2]
```diff
@@ -43,18 +44,30 @@ jobs:
...
+        run: |
+          nrn_branch=`echo $GITHUB_CONTEXT  \
+                      | jq -r '.event .pull_request .body' | grep "^CI_BRANCHES" \
+                      | awk -v FS="(NEURON_BRANCH=|,)" '{print $2}'`
```
Comment (olupton) [inline .github/workflows/test-as-submodule.yml:58]:
```suggestion
          nrn_branch=`echo ${GITHUB_PR_BODY} | grep "^CI_BRANCHES" \
                      | awk -v FS="(NEURON_BRANCH=|,)" '{print $2}'`
```

File: .github/workflows/test-as-submodule.yml [hunk 3]
```diff
@@ -29,10 +29,11 @@ jobs:
     env:
       CMAKE_BUILD_PARALLEL_LEVEL: ${{matrix.cores}}
     steps:
+
       - name: Install homebrew packages
         if: startsWith(matrix.os, 'macOS')
         run: |
-          brew install bison coreutils flex ninja openmpi
+          brew install bison coreutils flex ninja openmpi jq
```
Comment (olupton) [inline .github/workflows/test-as-submodule.yml:36]:
```suggestion
          brew install bison coreutils flex ninja openmpi
```

File: .github/workflows/test-as-submodule.yml [hunk 4]
```diff
@@ -43,18 +44,30 @@ jobs:
         run: |
           sudo apt-get install bison cython3 flex libfl-dev libopenmpi-dev \
             ninja-build openmpi-bin python3-dev python3-numpy python3-pytest \
-            python3-pytest-cov
+            python3-pytest-cov jq
```
Comment (olupton) [inline .github/workflows/test-as-submodule.yml:47]:
```suggestion
            python3-pytest-cov
```

**Review by olupton — APPROVED** [review]
LGTM!

I think the PR parsing can be done without `jq` too, but I don't feel strongly.
(the `GITHUB_PR_BODY` variable is probably not really necessary either, but I thought it might sidestep issues if there are quotes in the body.

**Comment by pramodk** [comment]
thanks! didn't realise gihub context can be accessed directly! :👌

**Review by olupton — APPROVED** [review]

**Pull request merged**